File: src/org/codepulsar/ast/expression/Grouping.java
package org.codepulsar.ast.expression;

import org.codepulsar.ast.Expression;

public class Grouping extends Expression {
    private final Expression expression;

    public Grouping(Expression expression) {
        this.expression = expression;
    }

    public <R> R accept(Visitor<R> visitor) {
        return visitor.visitGroupingExpression(this);
    }

    public Expression getExpression() {
        return this.expression;
    }
}

File: src/codepulsar/ast/expression/Grouping.cpp
#include "Grouping.h"


Pulsar::Grouping::Grouping(Expression* expression, int line) {
    this->expression = expression;
    this->line = line;
}

template<typename R>
R Pulsar::Grouping::accept(ExprVisitor<R>& visitor) {
    visitor.visitGroupingExpression(this);
}

Pulsar::Expression* Pulsar::Grouping::getExpression() {
    return this->expression;
}

int Pulsar::Grouping::getLine() {
    return this->line;
}
